### DISCUSSION TITLE
Question about langchain4j-google-ai-gemini

### DISCUSSION THREAD

**Discussion by Martin7-1** [langchain4j/langchain4j#1740] [Q&A] [answered]
After reading PR #1695, I'm curious about why `langchain4j-google-ai-gemini` create `GeminiService` in every request. Should it reuse and created in the constructor?

cc @glaforge

**Comment by glaforge** [comment, marked as answer]
Good question! And good point!
It's an oversight, I didn't really think about it.
I don't know if Retrofit is threadsafe... it that weren't the case, maybe it could make sense to keep reinstantiating a new service though.
But I like your suggestion and will likely make it once in the constructor and reuse it.

**Reply by Martin7-1** [reply]
Thank you!
